components/brave_wallet/browser/polkadot/polkadot_substrate_rpc.cc

```diff
@@ -40,6 +46,160 @@ net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+    return nullptr;
+  }
+  sv.remove_prefix(8);
+
+  account->nonce = base::ByteSwap(account->nonce);
+  account->consumers = base::ByteSwap(account->consumers);
+  account->providers = base::ByteSwap(account->providers);
+  account->sufficients = base::ByteSwap(account->sufficients);
+
+  account->data = mojom::PolkadotAccountBalance::New();
+
+  uint64_t low = 0, high = 0;
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &low)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  low = base::ByteSwap(low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &high)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  high = base::ByteSwap(high);
+
+  account->data->free = mojom::uint128Ptr(std::in_place, high, low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &low)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  low = base::ByteSwap(low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &high)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  high = base::ByteSwap(high);
+
+  account->data->reserved = mojom::uint128Ptr(std::in_place, high, low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &low)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  low = base::ByteSwap(low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &high)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  high = base::ByteSwap(high);
+
+  account->data->frozen = mojom::uint128Ptr(std::in_place, high, low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &low)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  low = base::ByteSwap(low);
+
+  if (!base::HexStringToUInt64(sv.substr(0, 16), &high)) {
+    return nullptr;
+  }
+  sv.remove_prefix(16);
+  high = base::ByteSwap(high);
+
+  account->data->flags = mojom::uint128Ptr(std::in_place, high, low);
+
+  return account;
+}
+
+mojom::PolkadotAccountInfoPtr ParseAccountInfoFromJson(
+    const std::optional<
+        std::vector<polkadot_substrate_rpc_responses::AccountInfo>>& result) {
+  /*
+
+  The shape of the resturned JSON is:
+
+  {
+    ...,
+    "result": [
+      {
+        "block": "<block hash>",
+        "changes": [["<storage key>", "<account info>" | null], ...]
+      },
+      ...
+    ]
+  }
+
+  */
+
+  if (!result) {
+    return nullptr;
+  }
+
+  auto& accounts = *result;
+  if (accounts.size() != 1) {
+    return nullptr;
+  }
+
+  auto& changes = accounts[0].changes;
+  if (changes.empty()) {
+    return nullptr;
+  }
+
+  auto* list = changes[0].GetIfList();
+  if (!list || list->size() != 2) {
+    return nullptr;
+  }
+
+  constexpr const char kFallback[] =
+      "0x0000000000000000000000000000000000000000000000000000000000000000"
+      "000000000000000000000000000000000000000000000000000000000000000000"
+      "000000000000000000000000000080";
+
+  std::string_view sv;
+  if (auto* str = (*list)[1].GetIfString()) {
+    sv = *str;
+  } else {
+    sv = kFallback;
+  }
+
```

Comment:
What is the prupose of kFallback info? Is this a legitimate response when not a string returned?

Reply:
If we have to return some account info, let's return an instance of PolkadotAccountInfoPtr filled with fallback values